tests/integration/sentinel/transitions/create-user-for-contacts.spec.js

```diff
@@ -183,68 +174,78 @@ describe('create_user_for_contacts', () => {
     assert.isTrue(newUserSettings.token_login.active);
     assert.match(newUserSettings._id, /^org\.couchdb\.user:new-person-\d\d\d\d/);
     assert.match(newUserSettings.name, /^new-person-\d\d\d\d$/);
-    // Login token sent
-    const queuedMsgs = await getQueuedMessages();
-    assert.lengthOf(queuedMsgs, 1);
-    assert.deepInclude(queuedMsgs[0], {
-      type: 'token_login',
-      user: newUserSettings._id
+
+    // New user created for additional contact
+    const [additionalUserSettings, ...moreUsers] = await utils.getUserSettings({ contactId: additionalPerson._id });
+    assert.isEmpty(moreUsers);
+    newUsers.push(additionalUserSettings.name);
+    assert.deepInclude(additionalUserSettings, {
+      roles: [additionalPerson.role],
+      phone: additionalPerson.phone,
+      facility_id: additionalPerson.parent._id,
+      contact_id: additionalPerson._id,
+      fullname: additionalPerson.name,
     });
-    assert.equal(queuedMsgs[0].tasks[0].messages[0].to, NEW_PERSON.phone);
+    assert.isTrue(additionalUserSettings.token_login.active);
+    assert.match(additionalUserSettings._id, /^org\.couchdb\.user:additional-person-\d\d\d\d/);
+    assert.match(additionalUserSettings.name, /^additional-person-\d\d\d\d$/);
 
-    // Other user still associated with original contact
-    const [otherUserSettings] = await utils.getUserSettings({ name: otherUser.username });
-    assert.equal(otherUserSettings.contact_id, ORIGINAL_PERSON._id);
-    // Can still log in as other user
-    assert.include(await loginAsUser(otherUser), { statusCode: 302 });
+    // Login tokens sent
+    const queuedMsgs = await messagesUtils.getQueuedMessages();
+    assert.lengthOf(queuedMsgs, 2);
+    queuedMsgs.forEach(msg => assert.equal(msg.type, 'token_login'));
+    const replacedUserMsg = queuedMsgs.find(msg => msg.user === newUserSettings._id);
+    assert.equal(replacedUserMsg.tasks[0].messages[0].to, NEW_PERSON.phone);
+    const additionalUserMsg = queuedMsgs.find(msg => msg.user === additionalUserSettings._id);
+    assert.equal(additionalUserMsg.tasks[0].messages[0].to, additionalPerson.phone);
```

Comment:
I think this needs a little bit of attention. 

As I said in a comment above, checking infodoc initial_replication_date vs last_replication_date will probably skip processing a lot of docs if sentinel is slower than edits come it. I changed the code to check if any transitions had run instead. 
So now I have this scenario: 

- Enable the transition and turn off sentinel. 
- I create a contact with phone number "1234"
- From the admin app I create a user for this contact, and set their phone number to "1234" as well. 
- Login with this user. 
- Submit a user replacement form. The new contact will also have phone number "1234". 
- Sync. 
- Start sentinel. 

What I'm seeing as a result is that two token login links were sent to "1234", one for the create and one for the replace. 
What is the user supposed to do in this situation? Which of the SMS-s is going to be correct?

Reply:
I am not sure if in your mind this situation is different now that we are not using initial/last_replication_date, but I think it is worth noting that the new user from the `replace` flow will only have the _same phone_ as the `create` user if the same phone number is used when creating the _new replacement contact_.  An alternative scenario would be that the new replacement contact is given a different phone and the token login links go to different phone numbers.  

We could add logic to the transition to only run both `create` and `replace` at the same time if they are going to different phone numbers, but that feels overly complex.  I would rather we just decide on an agreed upon default behavior and go with it:

1. Run both `create` and `replace`
2. Only run `create`
3. Only run `replace`
4. Run neither

None of these should be hard to implement.  Mokthar and I were originally thinking that `1.` was going to be least surprising, but if you prefer one of the other options, I am happy to go that direction.

Reply:
It's a difficult problem. 

I think that 1) produces the undesirable scenario that I described above, where a single phone number will get two SMSs at the same time, after they had been logged out, with only one of them being correct (to the replaced user) and the other one being to the created user.  
In this scenario, the user getting the SMS from the replace (the last action that was added) would have been correct. 

I would go for the last action that was added to the contact, if possible. This would mean that: replace removes create and create removes replace, so that only one of them (the last one) gets run. 

If that's not possible, I would just run the replace - especially since we're kind of designing this to only work on new contacts.

Reply:
> worth noting that the new user from the replace flow will only have the same phone as the create user if the same phone number is used when creating the new replacement contact. 

I used the forms that were provided in the tests/default config, the ones that create a new contact within the replace form. That form automatically copies the phone number from the replaced contact onto the new contact, does it not?

Reply:
Technically, the `replace` should always be the _last_ one that is set (since `create` should only be set when actually first writing the contact).  For the sake of simplicity, I have just updated the transition to only run `replace` if both are set.

Reply:
Excellent! 
Thanks a lot!